ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java

```diff
@@ -33,16 +32,14 @@
 public interface RatisMetricRegistry {
   Timer timer(String name);
 
-  Counter counter(String name);
+  LongCounter counter(String name);
 
   boolean remove(String name);
 
   <T> void gauge(String name, Supplier<Supplier<T>> gaugeSupplier);
 
   Timer timer(String name, MetricRegistry.MetricSupplier<Timer> supplier);
 
-  Counter counter(String name, MetricRegistry.MetricSupplier<Counter> supplier);
```

Comment:
While I don't find any calls in ratis code, I'm not sure if removing it directly would introduce compatibility issues.

Reply:
@codings-dan , thanks for reviewing this.

- First of all, ratis-metrics is not yet a public API.  It is neither in the ratis-server-api module nor in the org.apache.ratis.protocol package.
- The JIRA is to remove the use Counter.  There is no way to make it API compatible (e.g. Changing Counter to LongCounter is already API incompatible).  The user applications (e.g. Ozone) using it must update their code.

We may consider NOT merging this to 2.4.0 for now.  Once we have finished all the sub tasks in  RATIS-1688, we may merge all the changes to branch-2 and require the user applications to update their code.

Reply:
> * There is no way to make it API compatible
Got it. 

> We may consider NOT merging this to 2.4.0 for now
Sure, I also think 2.4.0 doesn't need to include RATIS-1688 related code.